src/goto-programs/goto_program_template.h

```diff
@@ -516,99 +478,60 @@ void goto_program_templatet<codeT, guardT>::compute_loop_numbers()
 }
 
 template <class codeT, class guardT>
-void goto_program_templatet<codeT, guardT>::get_successors(
-  targett target,
-  targetst &successors)
+template <typename Target>
+std::list<Target> goto_program_templatet<codeT, guardT>::get_successors(
+  Target target) const
 {
-  successors.clear();
   if(target==instructions.end())
-    return;
+    return std::list<Target>();
```

Comment:
By using early returns, we can restrict the scope of mutable state, which makes it easier to reason about the behaviour of this method.